common/core/core_session.c

```diff
@@ -441,11 +340,15 @@ static void start_request(const core_msg_t *core_msg) {
  *****************************************************************************/
 
 void core_session_clear_metadata() {
-  memzero(&session, sizeof(session_config_t));
+  memzero(&session, sizeof(session_private_t));
 }
 
 void core_session_parse_start_message(const core_msg_t *core_msg) {
   size_t request_type = core_msg->session_start.request.which_request;
+  // used cy_malloc() due to branched flow
+  session_ctx = (session_ctx_t *)cy_malloc(sizeof(session_ctx_t));
```

Comment:
I don't think we need `cy_malloc`, we can simple use `malloc` as the `session_ctx` should only be in scope for this function and there are no early exits.